contracts/libraries/CallbackValidation.sol

```diff
@@ -31,6 +31,6 @@ library CallbackValidation {
         returns (IUniswapV3Pool pool)
     {
         pool = IUniswapV3Pool(PoolAddress.computeAddress(factory, poolKey));
-        require(msg.sender == address(pool));
+        require(msg.sender == address(pool), 'Callback not called from pool');
```

Comment:
our general MO around error messages in core was to remove them if they're not expected to be hit in a normal case, and since this is only to protect against malicious behavior, perhaps not worth the gas? makes the new test that much less obvious, unfortunately

Reply:
this makes sense